tests/test_lint_recipe.py

```diff
@@ -582,9 +582,8 @@ def test_bad_requirements_order(self):
 
     def test_noarch_python_bound(self):
         expected_message = (
-            "noarch: python recipes are recommended to have a lower bound "
-            "on the python version. This recommendation will become a "
-            "requirement in the future."
+            "noarch: python recipes are required to have a lower bound "
+            "on the python version."
```

Comment:
How about explaining the solution a little more explicitly?
```suggestion
            "noarch: python recipes are required to have a lower bound "
            "on the python version. Typically this means putting "
            "`python >=3.6` in **both** `host` and `run` but you should check "
            "upstream for the package's Python compatibility."
```